### PR TITLE
USWDS - Search: Use consistent search markup.

### PR DESCRIPTION
**Pull request by mejiaj**
## Description

Search styles depend on `div[role="search"]` being a child of `.usa-search`. Otherwise it breaks to next line on small screens.

## Additional information
image

image

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [ ] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [ ] Run `npm test` and make sure the tests for the files you have changed have passed.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

**Comment by mejiaj** [comment]
@thisisdano, looks like axe is complaining about the search component page where the variations are listed on the same page.

I've tried adding them to their own iframe, but then axe doesn't test them. We could:
1. Separate into separate components and ignore the combined one.
2. Ignore the combined one and use the `search-header` component to test the accessibility of search

Happy to hear any other suggestions though.

**Comment by thisisdano** [comment]
I think it is a best practice to put the `role=search` in the form element, as we changed it in `2.5.0`, so I'd prefer to change the stylesheet to style properly whether or not the `role` is in the `< 2.5.0` markup or the `2.5.0+` markup, to support both better markup and backward compatibility.

I'll push this change.

File: src/components/search/search.njk [hunk 1]
```diff
@@ -14,22 +14,26 @@
   <h2>Big</h2>
 
   <form class="usa-search usa-search--big" role="search">
-    <label class="usa-sr-only" for="search-field-big">Search</label>
-    <input class="usa-input" id="search-field-big" type="search" name="search">
-    <button class="usa-button" type="submit">
-      <span class="usa-search__submit-text">Search</span>
-    </button>
+    <div>
```
Comment (mejiaj) [inline src/components/search/search.njk]: Looks like this in fractal for me:
image

We need to add the role to the `div`, remove the div, or update the styling.

File: src/stylesheets/components/_search.scss [hunk 2]
```diff
@@ -11,9 +11,11 @@
   @include typeset($theme-search-font-family);
   position: relative;
 
-  // if role is in the <form> element (>=2.5.0)...
+  // if role is in the <form> element (>=2.6.0)...
   &[role="search"],
-  // ...or if the component has a separate <div> (<2.5.0)
+  // if there is a (=2.5.[0,1])...
+  &[role="search"] div,
```
Comment (mejiaj) [inline src/stylesheets/components/_search.scss]: Should we be extra-cautious and do `&[role="search"] > div`?

Comment (thisisdano) [reply src/stylesheets/components/_search.scss]: Yes

Comment (thisisdano) [reply src/stylesheets/components/_search.scss]: I'll do that

**Pull request merged**